test/api_test.js

```diff
@@ -569,6 +622,105 @@ describe('Api', () => {
     });
   });
 
+  describe('credentials', () => {
+    const P = f => path.resolve(__dirname, f);
+
+    const loadFile = (filename) => {
+      return fs.readFileSync(P(filename)).toString('base64');
+    };
+
+    const captureError = async(fn, args) => {
+      try {
+        await fn.apply(client, args);
+        return 'Should fail';
+      } catch (err) {
+        return err.code;
+      }
+    };
+
+    const expectedError = 'AuthorizationFailed';
+    const createTemporaryCredentials = taskcluster.createTemporaryCredentials;
+
+    before(() => {
+      taskcluster.createTemporaryCredentials = sandbox.stub();
+      taskcluster.createTemporaryCredentials.returns({
+        clientId: 'myClientId',
+        accessToken: 'myAccessToken',
+        certificate: 'myCertificate',
+      });
+    });
+
+    after(() => {
+      taskcluster.createTemporaryCredentials = createTemporaryCredentials;
+    });
+
+    it('invalid signature', async() => {
+      const errorCode = await captureError(client.getCredentials, [{
+        signature: pkcs7.invalid,
+      }]);
+      assume(errorCode).eql(expectedError);
+    });
+
+    it('instance not in the database', async() => {
+      const errorCode = await captureError(client.getCredentials, [{
+        signature: pkcs7.signedData,
+      }]);
+      assume(errorCode).eql(expectedError);
+    });
+
+    it('instance is not running', async() => {
+      await state.insertInstance({
+        id: pkcs7.doc.instanceId,
+        workerType,
+        region: pkcs7.doc.region,
+        instanceType,
+        state: 'pending',
+        az,
+        imageId,
+        launched,
+        lastEvent: new Date(),
+      });
+
+      const errorCode = await captureError(client.getCredentials, [{
+        signature: pkcs7.signedData,
+      }]);
+      assume(errorCode).eql(expectedError);
+    });
+
+    it('claimed successfully', async() => {
+      await state.insertInstance({
+        id: pkcs7.doc.instanceId,
+        workerType,
+        region: pkcs7.doc.region,
+        instanceType,
+        state: 'running',
+        az,
+        imageId,
+        launched,
+        lastEvent: new Date(),
+      });
+
+      const result = await client.getCredentials({
+        signature: pkcs7.signedData,
+      });
+      assume(result).eql({
+        clientId: 'myClientId',
+        accessToken: 'myAccessToken',
+        certificate: 'myCertificate',
+      });
+
+      /*await client.claimCredentials({
```

Comment:
We should probably be calling this function for real in the test.  Is there any reason why it's commented out?  The whole commented out block is stuff that we must test.

Reply:
Good catch, I commented for a test and forgot to uncomment it